File: bcc/bcc_python/hello.c
/**
 * @file hello.c
 * @author your name (you@domain.com)
 * @brief   使用C写一个eBPF程序
 * @version 0.1
 * @date 2022-03-21
 * 
 * @copyright Copyright (c) 2022
 * 
 */

int  hello_world(void *ctx)
{
    bpf_trace_printk("Hello world!");
    return 0;
}

/**
 * 1、bpf_trace_printk 是 BPF内核函数，作用是输出字符串，在内核事件触发时，将字符串"Hello world"输出到Linux的调试文件中
 * /sys/kernel/debug/tracing/trace_pipe
 *
 */

File: bcc/bcc_python/hello.py
#!/usr/bin/env python3

# 1、引入bcc 库的BPF 模块
from  bcc  import BPF
# 2、调用BPF() 加载BPF 源码
b = BPF(src_file="hello.c")

# 3、将BPF程序挂载到 内核探针 kprobe  内核函数do_sys_openat2(),触发函数时执行钩子函数  hello_word
b.attach_kprobe(event="do_sys_openat2", fn_name="hello_world")
# 4) read and print /sys/kernel/debug/tracing/trace_pipe
b.trace_print()

